### DISCUSSION TITLE
Is custom HttpClientAdapter implementation required for TLS certificate validation?

### DISCUSSION THREAD

**Discussion by mplgn** [cfug/dio#2307] [Q&A] [answered]
Hi, Flutter and dio newbie question:   Does dio automatically perform TLS certificate validation, or do I need to implement a custom HttpClientAdapter as shown here [https://pub.dev/packages/dio#https-certificate-verification](https://pub.dev/packages/dio#https-certificate-verification) ?   I'm unclear if this is required, or only necessary to customize default validation.   Thanks.

**Comment by kuhnroyal** [comment, marked as answer]
This happens automatically for certificates where the Root CA is considered valid by the device.
You only need to implement custom certificate handling if you want to use custom or self signed certs.

**Reply by mplgn** [reply]
Great, thank you.
